<a href="https://colab.research.google.com/github/estefayjuanma1/MiniProy1_Senales3/blob/master/Avance1MiniProy1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Mini-Proyecto 1: Análisis de bio-señales para computación afectiva.** 
## **Tratamiento de señales III**

*Carolina Bedoya Doria* 
*C.C:* 

*Estefania Torres Zuluaga*
*C.C: 1039471113*

*David Yepes*
*C.C:*


##Primer acercamiento al Dataset



*   Se entrevistaron 32 personas
*   Cada una de las personas usadas para el experimento vió 40 videos (de 1 min cada uno), para lograr identificar las emociones o sensaciones que causaba en ellos. 
*   Se tienen entonces 40 canales con los diferentes "sensores" o electrodos con los que se intervino a cada persona, distribuídos así:

      * **1-32:** Se destinan 32 canales para medir señales electroencefalográficas (EEG), las cuales nos permiten detectar la actividad eléctrica del cerebro
      * **33-34:** Dos canales para realizar mediciones electro-oculográficas (EOG), es decir, los movimientos oculares; uno de ellos para los movimientos verticales y otro para los horizontales
      * **35 - 36:** Destinados a medir señales de electromiogramas (EMG), los cuales, para este caso, nos permiten identificar los movimientos de los músculos trapecio (ubicado en la espalda) y zigomático (debajo del ojo)
      * **37:** Un canal destinado a medir la "respuesta galvánica de la piel" (GSR), es decir, la medida de las variaciones en las características eléctricas de la piel (como por ejemplo la conductancia, causada por la sudoración)
      * **38:** Un canal usado para medir la frecuencia respiratoria a través de un "cinturón" o banda
      * **39:**Un Canal que usa un pletismógrafo para medir cambios de volumen en el cuerpo por medio del flujo sanguíneo e incluso para medir el ritmo cardiaco
      * **40:**Un último canal para medir la temperatura de la piel 

*  Para realizar el estudio o "clasificación" de las diferentes emociones que induce cada uno de los videos a las diferentes personas, se definen entonces 4 criterios o etiquetas, todos con una escala del 1 al 9:
      * Exitación (o arousal):
      * Valencia: 

*  Cada uno de los sujetos estudiados,



##Contexto: 
Con el presente proyecto se busca construir una especie de framework que nos permita procesar y caracterizar las señales fisiológicas, para luego agruparlas en
estados emocionales. Para  lograr dicho objetivo, partiremos del conjunto de datos del DEAP, el cual se basa en la coloación de diferentes electrodos y sensores en el cuerpo humano, para medir los cambios o variaciones que le ocurren a cada persona al visualizar diferentes tipos de video (movimientos oculares, ondas cerebrales, sudoración, temperatura de la piel, entre otros). Con ello, se tendrá entonces un conjunto de señales fisiológicas, con las cuales se prodrá realizar un adecuado procesamiento y análisis de cada una de ellas, para así lograr caracterizarlas. 

Para realizar la caracterización de las señales, se escogerán 3 características por cada tipo de señal obtenida de la base de datos (EEG, EOG, EMG, GSR Frecuencia respiratoria, Flujo sanguíneo y Temperatura), como se muestra a continuación: 

*  **EEG:**Teniendo en cuenta que los electroencefalogramas nos ayudan a registrar la actividad eléctrica del cerebro, procederemos a conceptualizar cada una de las bandas espectrales presentadas en las EEG, para lograr una  mayor claridad frente a la escogencia de las caracteristicas mas relevantes de dicha señal. 
      * **Banda Delta:** Definidas hasta los 3,5 Hz.  Son ondas de baja frecuencia y alta intensidad (centenares de μV). Tienen lugar en niños de corta edad y en adultos únicamente en estado de sueño profundo, inconsciencia o situaciones que aumente la presión intracraneal como tumores
cerebrales.
      * **Banda Theta:**De los 3,5 a 7,5 Hz. Estas ondas de amplitud inferior a 20 μV se dan durante el proceso de maduración en toda la corteza cerebral, aunque predominan en la región occipital y temporal y es más rápida en la zona frontal. Es clasificada como "actividad lenta", se presenta con normalidad en niños de hasta 14 años y anormalidad en adultos despiertos. Puede verse como una manifestación de lesiones o trastornos difusos. 
      * **Banda Alfa:** Definida entre los 7.5 y 12.5 Hz. Término usado para ráfagas de 20-100 μV de amplitud y gran
periodicidad. Estas ondas se asocian a periodos de relajación, de inactividad, y son muy patentes en ausencia de estímulos visuales (cuando se cierran los ojos).Se desaparece al abrir los ojos o alertar por cualquier mecanismo (pensar, calcular).
Existe mucha variabilidad interpersonal en el ritmo alfa. Hay personas sin ritmo alfa y otras con ritmo alfa permanente, aunque suele observarse regularmente en adultos relajados y es una de las ondas  presentes la mayor parte de la vida (después de los 13 años)
      * **Banda Beta:**de 12,5 a 30 Hz. Son señales de muy baja amplitud, por debajo de 20 μV, son bastante comunes y predominan durante la edad adulta. Suelen subdividirse en beta baja (suele localizar en los lóbulos frontal y occipital), media y alta (menos localizados).
Más irregular que el ritmo alfa, se asocia a actividad psicofísica, estados de agitación, alerta, ansiedad o la actividad mental que se realiza en la resolución de problemas.
      * **Banda Gamma:** Se referiere al rango de frecuencias de 30 a 50 Hz 
Esta banda tiene un pico de resonancia, similar a los ritmos alfa, cercano a los 40Hz al que se le da el nombre de épsilon, que se asocia a la actividad mental abstracta que interviene en percibir como único, por ejemplo, el olor, los rasgos de la cara, la personalidad y la voz de una persona.

  Con las anteriores definiciones, tenemos una base para caracterizar nuestra actividad cerebral, para la cual se tendrá en cuenta:
      * Densidad espectral de potencia (DEP) de las ondas teta de los correspondientes pares de electrodos
      * DEP de las ondas alfa de los correspondientes pares de electrodos
      * DEP de las ondas beta de los correspondientes pares de electrodos
      * DEP de las ondas gama de los correspondientes pares de electrodos

*  **EOG:** Para las señales electr-oculográficas (movimientos oculares tanto verticales como horizontales) tendremos en cuenta entonces:
    * Media y varianza
    * Energía
    * Frecuencia de parpadeo de los ojos

*  **EMG:**Con el objetivo de caracterizar los movimientos de los músculos trapecio y zigomático se tendrá en cuenta:
    * Media
    * Varianza
    * Energía

*  **GSR:** Para medir las variaciones de las características eléctricas de la piel se tendrá en cuenta:
    * Resistencia promedio de la piel
    * Promedio de las derivadas
    * Promedio de las derivadas únicamente para valores negativos
    * Número de mínimos locales
    *
*  **Patrón de respiración:** Con  la banda que mide la frecuencia respiratoria, prodeceremos a extraer:
    * Señal de respiración promedio
    * Variación de la respiración (promedio de sus derivadas)
    * Ritmo respiratorio (centroide espectral)
    * Frecuencia respiratoria
    * 10 potencias espectrales en la banda de 0 a 2,4 Hz

*  **Presión sanguínea (vol de flujo sanguíneo):** El pletismógrafo nos permitirá medir el flujo sanguíneo, el cual a su vez no da una idea de como estaba variando el ritmo cardiaco. Para ello se exraerán las siguientes características:
    * Valor promedio y desviación estándar del ritmo cardiaco
    * Intervalos entre latidos
    * DEP de diferentes bandas de frecuencia
*  **Temperatura de la piel:**Para medir la señal de temperatura se procederá a extraer:
    * Valor promedio
    * Valor promedio de sus derivadas
    * Potencia espectral en la banda de 0 a 0,1 Hz y 0,1 a 0,2 Hz



Finalmente, con las características a tener en cuenta para cada una de las señales presentadas, se procederá a condensarlas en una misma tabla con las respectivas etiquetas emocionales según los niveles de exitación y valencia, para tener finalmente un buen análisis de los datos 


## EXTRACCIÓN DE CARACTERÍSTICAS


In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [ ]:
import scipy.io as sio
import matplotlib.pyplot as plt
import numpy as np
from scipy.signal import find_peaks

#Cargamos una parte de la base de datos (los 40 videos vistos por el sujeto #17)
ruta = '/content/drive/My Drive/Miniproyecto/'
fileName = 's17.mat'
db = sio.loadmat(ruta+fileName)
db

{'__globals__': [],
 '__header__': b'MATLAB 5.0 MAT-file, Platform: PCWIN64, Created on: Wed Aug 21 08:32:38 2013',
 '__version__': '1.0',
 'data': array([[[-4.74806162e+00,  1.25138383e+01, -2.42521931e+00, ...,
          -9.61267515e+00,  1.05404439e+01,  3.97091891e+01],
         [-5.23313973e+00,  1.14957796e+01, -7.74271761e+00, ...,
          -4.69191526e+01, -3.57270098e+01, -5.87687115e+00],
         [-4.74208077e+00,  1.20123950e+01, -2.66537787e+00, ...,
           1.28023839e+01,  3.40084793e+01,  6.55370632e+01],
         ...,
         [-6.15474196e+03, -6.17105443e+03, -6.25352303e+03, ...,
           1.28883803e+03,  2.73371158e+02,  1.11046336e+03],
         [-4.29735518e+03, -4.41307608e+03, -4.51854323e+03, ...,
           6.74837536e+03,  6.66732191e+03,  6.53744107e+03],
         [-6.53897870e-03, -6.53897870e-03, -6.53897870e-03, ...,
          -2.43535942e-01, -2.43535942e-01, -2.43535942e-01]],
 
        [[-1.25487623e+01, -1.16982806e+01, -1.57186565e+01, ...,
  

In [ ]:
X = db['data']
X.shape

(40, 40, 6528)

Cada arreglo X, el cual me representa a cada sujeto tendrá entonces la forma de (40, 40, 6528) los cuales me representan respectivamente:
* Las 40 respuestas emocionales producidas por los 40 videos
* Las 40 señales fisológicas medidas
* Los datos obtenidos en la realización del experimento (removiendo los primeros 3 segundos que fueron de acondicionamiento de las diferentes personas)

##Importando librerías

In [ ]:

from scipy import signal
from scipy.signal import find_peaks
from scipy.fft import fft
import matplotlib.pyplot as plt
import pandas as pd


##Definiendo funciones generales

In [ ]:

def media(data):
  media= np.mean(data)
  #print('Media =',media)
  return media

def varianza(data):
  varianza= np.var(data)
  #print('Varianza =',varianza)
  return varianza

def energia(señal):
  energia= np.sum(np.power(señal,2))
  #print('Energía =',energia)
  return energia

def media_deriv(data):
  derivada = np.diff(data)
  media= np.mean(derivada)
  #plt.plot(derivada)
  #print('Derivada promedio =',media)
  return media

def desvi_estand(data):
  desvi= np.std([data], ddof=1)
  #print('Desviación Estandar =',desvi)
  return desvi

def media_deri_neg(data):
  n=[]
  derivada = np.diff(data)
  for i in derivada:
    if i < 0:
      n=[i]
      median= np.mean(n)
  return median

def psd(data):
  fs = 128 # fs
  f, Pxx_fil = signal.welch(data, fs, nperseg=64)
  #plt.semilogy(f, Pxx_fil)
  #plt.ylim([0.5e-3, 1])
  #plt.xlabel('frequency [Hz]')
  #plt.ylabel('PSD [V**2/Hz]')
  #plt.show()
  return (Pxx_fil)  
  
def potencia(data):
  pot = np.sum(data)
  #print("P",pot)
  return pot

def lowpass_filter(data, fcorte, fs, orden):
    nyq = 0.5*fs
    fcorte_nor = fcorte / nyq
    # Get the filter coefficients 
    b, a = signal.butter(orden, fcorte_nor, btype='low', analog=False)
    y = signal.filtfilt(b, a, data)
    return y

def highpass_filter(data, fcorte, fs, orden):
    nyq = 0.5*fs
    fcorte_nor = fcorte / nyq
    # Get the filter coefficients 
    b, a = signal.butter(orden, fcorte_nor, btype='high', analog=False)
    y = signal.filtfilt(b, a, data)
    return y

def passband_filter(data, fmin, fmax, fs, orden):
    nyq = 0.5*fs
    fmin_nor = fmin / nyq
    fmax_nor = fmax / nyq
    b, a = signal.butter(orden, [fmin_nor, fmax_nor], btype='band', analog=False)
    y = signal.filtfilt(b, a, data)
    return y

def filtro_soslp(data, orden, fc):
  sos = signal.butter(orden, fc, 'lp', fs=128, output='sos')
  filtered = signal.sosfilt(sos, data)
  #plt.plot(time, filtered)
  #plt.title('Banda 0-0.1Hz')
  #plt.xlabel('Time [seconds]')
  #plt.show()
  return filtered

def filtro_sospb(data, orden, fcmin, fcmx):
  sos = signal.butter(6, [fcmin, fcmx], 'bp', fs=128, output='sos')
  filtered2 = signal.sosfilt(sos, data)
  #plt.plot(time, filtered2)
  #plt.title('Banda entre 0.1Hz y 0.2Hz')
  #plt.xlabel('Time [seconds]')
  #plt.show()
  return filtered2  


def tasa(data):
  peaks, _ = find_peaks(data, prominence=np.mean(ojov))
  np.diff(peaks)
  #plt.plot(data)
  #plt.plot(peaks, data[peaks], "x")
  #plt.show()
  return (len(peaks))


##Extracción de características EEG

In [ ]:
#Extrayendo cada uno de los canales de la EEG

#trial = 15 # video u observación
p1t=[]
p1a=[]
p1b=[]
p1g=[]


p2t=[]
p2a=[]
p2b=[]
p2g=[]

p3t=[]
p3a=[]
p3b=[]
p3g=[]

p4t=[]
p4a=[]
p4b=[]
p4g=[]

p5t=[]
p5a=[]
p5b=[]
p5g=[]

p6t=[]
p6a=[]
p6b=[]
p6g=[]

p7t=[]
p7a=[]
p7b=[]
p7g=[]

p8t=[]
p8a=[]
p8b=[]
p8g=[]

p9t=[]
p9a=[]
p9b=[]
p9g=[]

p10t=[]
p10a=[]
p10b=[]
p10g=[]

p11t=[]
p11a=[]
p11b=[]
p11g=[]

p12t=[]
p12a=[]
p12b=[]
p12g=[]

p13t=[]
p13a=[]
p13b=[]
p13g=[]


for i in range (40):
  channel = 1 # canal de la señal EEG FP1
  FP1 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta1= passband_filter(FP1, 4,7, 128, 4)
  Alfa1= passband_filter(FP1, 8,12, 128, 4)
  Beta1= passband_filter(FP1, 13,30, 128, 4)
  Gama1= highpass_filter(FP1, 30, 128, 4)

  channel = 2 # canal de la señal EEG AF3
  AF3 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta2= passband_filter(AF3, 4,7, 128, 4)
  Alfa2= passband_filter(AF3, 8,12, 128, 4)
  Beta2= passband_filter(AF3, 13,30, 128, 4)
  Gama2= highpass_filter(AF3, 30, 128, 4)

  channel = 3 # canal de la señal EEG F3
  F3 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta3= passband_filter(F3, 4,7, 128, 4)
  Alfa3= passband_filter(F3, 8,12, 128, 4)
  Beta3= passband_filter(F3, 13,30, 128, 4)
  Gama3= highpass_filter(F3, 30, 128, 4)

  channel = 4 # canal de la señal EEG F7
  F7 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta4= passband_filter(F7, 4,7, 128, 4)
  Alfa4= passband_filter(F7, 8,12, 128, 4)
  Beta4= passband_filter(F7, 13,30, 128, 4)
  Gama4= highpass_filter(F7, 30, 128, 4)

  channel = 5 # canal de la señal EEG FC5
  FC5 = X[-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta5= passband_filter(FC5, 4,7, 128, 4)
  Alfa5= passband_filter(FC5, 8,12, 128, 4)
  Beta5= passband_filter(FC5, 13,30, 128, 4)
  Gama5= highpass_filter(FC5, 30, 128, 4)

  channel = 6 # canal de la señal EEG F3
  FC1 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta6= passband_filter(FC1, 4,7, 128, 4)
  Alfa6= passband_filter(FC1, 8,12, 128, 4)
  Beta6= passband_filter(FC1, 13,30, 128, 4)
  Gama6= highpass_filter(FC1, 30, 128, 4)

  channel = 7 # canal de la señal EEG F4
  C3= X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta7= passband_filter(C3, 4,7, 128, 4)
  Alfa7= passband_filter(C3, 8,12, 128, 4)
  Beta7= passband_filter(C3, 13,30, 128, 4)
  Gama7= highpass_filter(C3, 30, 128, 4)

  channel = 8 # canal de la señal EEG F3
  T7 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta8= passband_filter(T7, 4,7, 128, 4)
  Alfa8= passband_filter(T7, 8,12, 128, 4)
  Beta8= passband_filter(T7, 13,30, 128, 4)
  Gama8= highpass_filter(T7, 30, 128, 4)

  channel = 9 # canal de la señal EEG F4
  CP5 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta9= passband_filter(CP5, 4,7, 128, 4)
  Alfa9= passband_filter(CP5, 8,12, 128, 4)
  Beta9= passband_filter(CP5, 13,30, 128, 4)
  Gama9= highpass_filter(CP5, 30, 128, 4)

  channel = 10 # canal de la señal EEG F3
  CP1 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta10= passband_filter(CP1, 4,7, 128, 4)
  Alfa10= passband_filter(CP1, 8,12, 128, 4)
  Beta10= passband_filter(CP1, 13,30, 128, 4)
  Gama10= highpass_filter(CP1, 30, 128, 4)

  channel = 11 # canal de la señal EEG F4
  P3 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta11= passband_filter(P3, 4,7, 128, 4)
  Alfa11= passband_filter(P3, 8,12, 128, 4)
  Beta11= passband_filter(P3, 13,30, 128, 4)
  Gama11= highpass_filter(P3, 30, 128, 4)

  channel = 12 # canal de la señal EEG F3
  P7 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta12= passband_filter(P7, 4,7, 128, 4)
  Alfa12= passband_filter(P7, 8,12, 128, 4)
  Beta12= passband_filter(P7, 13,30, 128, 4)
  Gama12= highpass_filter(P7, 30, 128, 4)

  channel = 13 # canal de la señal EEG F4
  PO3 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta13= passband_filter(PO3, 4,7, 128, 4)
  Alfa13= passband_filter(PO3, 8,12, 128, 4)
  Beta13= passband_filter(PO3, 13,30, 128, 4)
  Gama13= highpass_filter(PO3, 30, 128, 4)

  channel = 14 # canal de la señal EEG F3
  O1 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta14= passband_filter(O1, 4,7, 128, 4)
  Alfa14= passband_filter(O1, 8,12, 128, 4)
  Beta14= passband_filter(O1, 13,30, 128, 4)
  Gama14= highpass_filter(O1, 30, 128, 4)

  channel = 15 # canal de la señal EEG F4
  OZ = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta15= passband_filter(OZ, 4,7, 128, 4)
  Alfa15= passband_filter(OZ, 8,12, 128, 4)
  Beta15= passband_filter(OZ, 13,30, 128, 4)
  Gama15= highpass_filter(OZ, 30, 128, 4)

  channel = 16 # canal de la señal EEG F3
  PZ = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta16= passband_filter(PZ, 4,7, 128, 4)
  Alfa16= passband_filter(PZ, 8,12, 128, 4)
  Beta16= passband_filter(PZ, 13,30, 128, 4)
  Gama16= highpass_filter(PZ, 30, 128, 4)

  channel = 17 # canal de la señal EEG F4
  FP2= X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta17= passband_filter(FP2, 4,7, 128, 4)
  Alfa17= passband_filter(FP2, 8,12, 128, 4)
  Beta17= passband_filter(FP2, 13,30, 128, 4)
  Gama17= highpass_filter(FP2, 30, 128, 4)

  channel = 18 # canal de la señal EEG F3
  AF4 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta18= passband_filter(AF4, 4,7, 128, 4)
  Alfa18= passband_filter(AF4, 8,12, 128, 4)
  Beta18= passband_filter(AF4, 13,30, 128, 4)
  Gama18= highpass_filter(AF4, 30, 128, 4)

  channel = 19 # canal de la señal EEG F4
  FZ = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta19= passband_filter(FZ, 4,7, 128, 4)
  Alfa19= passband_filter(FZ, 8,12, 128, 4)
  Beta19= passband_filter(FZ, 13,30, 128, 4)
  Gama19= highpass_filter(FZ, 30, 128, 4)

  channel = 20 # canal de la señal EEG F3
  F4 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta20= passband_filter(F4, 4,7, 128, 4)
  Alfa20= passband_filter(F4, 8,12, 128, 4)
  Beta20= passband_filter(F4, 13,30, 128, 4)
  Gama20= highpass_filter(F4, 30, 128, 4)

  channel = 21 # canal de la señal EEG F4
  F8 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta21= passband_filter(F8, 4,7, 128, 4)
  Alfa21= passband_filter(F8, 8,12, 128, 4)
  Beta21= passband_filter(F8, 13,30, 128, 4)
  Gama21= highpass_filter(F8, 30, 128, 4)

  channel = 22 # canal de la señal EEG F3
  FC6 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta22= passband_filter(FC6, 4,7, 128, 4)
  Alfa22= passband_filter(FC6, 8,12, 128, 4)
  Beta22= passband_filter(FC6, 13,30, 128, 4)
  Gama22= highpass_filter(FC6, 30, 128, 4)

  channel = 23 # canal de la señal EEG F4
  FC2 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta23= passband_filter(FC2, 4,7, 128, 4)
  Alfa23= passband_filter(FC2, 8,12, 128, 4)
  Beta23= passband_filter(FC2, 13,30, 128, 4)
  Gama23= highpass_filter(FC2, 30, 128, 4)

  channel = 24 # canal de la señal EEG F3
  CZ = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta24= passband_filter(CZ, 4,7, 128, 4)
  Alfa24= passband_filter(CZ, 8,12, 128, 4)
  Beta24= passband_filter(CZ, 13,30, 128, 4)
  Gama24= highpass_filter(CZ, 30, 128, 4)

  channel = 25 # canal de la señal EEG F4
  C4 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta25= passband_filter(C4, 4,7, 128, 4)
  Alfa25= passband_filter(C4, 8,12, 128, 4)
  Beta25= passband_filter(C4, 13,30, 128, 4)
  Gama25= highpass_filter(C4, 30, 128, 4)

  channel = 26 # canal de la señal EEG F3
  T8 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta26= passband_filter(T8, 4,7, 128, 4)
  Alfa26= passband_filter(T8, 8,12, 128, 4)
  Beta26= passband_filter(T8, 13,30, 128, 4)
  Gama26= highpass_filter(T8, 30, 128, 4)

  channel = 27 # canal de la señal EEG F4
  CP6 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta27= passband_filter(CP6, 4,7, 128, 4)
  Alfa27= passband_filter(CP6, 8,12, 128, 4)
  Beta27= passband_filter(CP6, 13,30, 128, 4)
  Gama27= highpass_filter(CP6, 30, 128, 4)

  channel = 28 # canal de la señal EEG F3
  CP2 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta28= passband_filter(CP2, 4,7, 128, 4)
  Alfa28= passband_filter(CP2, 8,12, 128, 4)
  Beta28= passband_filter(CP2, 13,30, 128, 4)
  Gama28= highpass_filter(CP2, 30, 128, 4)

  channel = 29 # canal de la señal EEG F4
  P4 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta29= passband_filter(P4, 4,7, 128, 4)
  Alfa29= passband_filter(P4, 8,12, 128, 4)
  Beta29= passband_filter(P4, 13,30, 128, 4)
  Gama29= highpass_filter(P4, 30, 128, 4)

  channel = 30 # canal de la señal EEG F3
  P8 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta30= passband_filter(P8, 4,7, 128, 4)
  Alfa30= passband_filter(P8, 8,12, 128, 4)
  Beta30= passband_filter(P8, 13,30, 128, 4)
  Gama30= highpass_filter(P8, 30, 128, 4)

  channel = 31 # canal de la señal EEG F4
  PO4 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta31= passband_filter(PO4, 4,7, 128, 4)
  Alfa31= passband_filter(PO4, 8,12, 128, 4)
  Beta31= passband_filter(PO4, 13,30, 128, 4)
  Gama31= highpass_filter(PO4, 30, 128, 4)

  channel = 32 # canal de la señal EEG F3
  O2 = X[i-1,channel-1,:]
  #Delta1= passband_filter(AF3, 0.5, 4, 128, 4)
  Teta32= passband_filter(O2, 4,7, 128, 4)
  Alfa32= passband_filter(O2, 8,12, 128, 4)
  Beta32= passband_filter(O2, 13,30, 128, 4)
  Gama32= highpass_filter(O2, 30, 128, 4)

  #Calculando las asimetrías de cada par de electrodos
  TassAF3_AF4 = Teta2 - Teta18
  AassAF3_AF4 = Alfa2 - Alfa18
  BassAF3_AF4 = Beta2 - Beta18
  GassAF3_AF4 = Gama2 - Gama18

  TassF7_F8 = Teta4 - Teta21
  AassF7_F8 = Alfa4 - Alfa21
  BassF7_F8 = Beta4 - Beta21
  GassF7_F8 = Gama4 - Gama21

  TassF3_F4 = Teta3 - Teta20
  AassF3_F4 = Alfa3 - Alfa20
  BassF3_F4 = Beta3 - Beta20
  GassF3_F4 = Gama3 - Gama20

  TassFC5_FC6 = Teta5 - Teta22
  AassFC5_FC6 = Alfa5 - Alfa22
  BassFC5_FC6 = Beta5 - Beta22
  GassFC5_FC6 = Gama5 - Gama22

  TassT7_T8 = Teta8 - Teta26
  AassT7_T8 = Alfa8 - Alfa26
  BassT7_T8 = Beta8 - Beta26
  GassT7_T8 = Gama8 - Gama26

  TassP7_P8 = Teta12 - Teta30
  AassP7_P8 = Alfa12 - Alfa30
  BassP7_P8 = Beta12 - Beta30
  GassP7_P8 = Gama12 - Gama30

  TassO1_O2 = Teta14 - Teta32
  AassO1_O2 = Alfa3- - Alfa20
  BassO1_O2 = Beta3 - Beta20
  GassO1_O2 = Gama3 - Gama20


  TassFC5_P7 = Teta5 - Teta12
  AassFC5_P7 = Alfa5 - Alfa12
  BassFC5_P7 = Beta5 - Beta12
  GassFC5_P7 = Gama5 - Gama12

  TassFC6_P8 = Teta22 - Teta30
  AassFC6_P8 = Alfa22 - Alfa30
  BassFC6_P8 = Beta22 - Beta30
  GassFC6_P8 = Gama22 - Gama30

  TassP7_O1 = Teta12 - Teta14 
  AassP7_O1 = Alfa12 - Alfa14 
  BassP7_O1 = Beta12 - Beta14 
  GassP7_O1 = Gama12 - Gama14 

  TassP8_O2 = Teta30 - Teta32 
  AassP8_O2 = Alfa30 - Alfa32 
  BassP8_O2 = Beta30 - Beta32 
  GassP8_O2 = Gama30 - Gama32

  TassT7_O1 = Teta8 - Teta14
  AassT7_O1 = Alfa8 - Alfa14
  BassT7_O1 = Beta8 - Beta14
  GassT7_O1 = Gama8 - Gama14 

  TassT8_O2 = Teta26 - Teta32
  AassT8_O2 = Alfa26 - Alfa32
  BassT8_O2 = Beta26 - Beta32
  GassT8_O2 = Gama26 - Gama32

  #Sacando la DEP de los pares de electrodos
  tAF3_AF4 = psd(TassAF3_AF4)
  p1t.append(potencia(tAF3_AF4))
  aAF3_AF4 = psd(AassAF3_AF4)
  p1a.append(potencia(aAF3_AF4))
  bAF3_AF4 = psd(BassAF3_AF4)
  p1b.append(potencia(bAF3_AF4))
  gAF3_AF4 = psd(GassAF3_AF4)
  p1g.append(potencia(gAF3_AF4))

  tF7_F8 = psd(TassF7_F8)
  p2t.append(potencia(tF7_F8))
  aF7_F8 = psd(AassF7_F8)
  p2a.append(potencia(aF7_F8))
  bF7_F8 = psd(BassF7_F8)
  p2b.append(potencia(bF7_F8))
  gF7_F8 = psd(GassF7_F8)
  p2g.append(potencia(gF7_F8))

  tF3_F4 = psd(TassF3_F4)
  p3t.append(potencia(tF3_F4))
  aF3_F4 = psd(AassF3_F4)
  p3a.append(potencia(aF3_F4))
  bF3_F4 = psd(BassF3_F4)
  p3b.append(potencia(bF3_F4))
  gF3_F4 = psd(GassF3_F4)
  p3g.append(potencia(gF3_F4))

  tFC5_FC6 = psd(TassFC5_FC6)
  p4t.append(potencia(tFC5_FC6))
  aFC5_FC6 = psd(AassFC5_FC6)
  p4a.append(potencia(aFC5_FC6))
  bFC5_FC6 = psd(BassFC5_FC6)
  p4b.append(potencia(bFC5_FC6))
  gFC5_FC6 = psd(GassFC5_FC6)
  p4g.append(potencia(gFC5_FC6))

  tT7_T8 = psd(TassT7_T8)
  p5t.append(potencia(tT7_T8))
  aT7_T8 = psd(AassT7_T8)
  p5a.append(potencia(aT7_T8))
  bT7_T8 = psd(BassT7_T8)
  p5b.append(potencia(bT7_T8))
  gT7_T8 = psd(GassT7_T8)
  p5g.append(potencia(gT7_T8))

  tP7_P8 = psd(TassP7_P8)
  p6t.append(potencia(tP7_P8))
  aP7_P8 = psd(AassP7_P8)
  p6a.append(potencia(aP7_P8))
  bP7_P8 = psd(BassP7_P8)
  p6b.append(potencia(bP7_P8))
  gP7_P8 = psd(GassP7_P8)
  p6g.append(potencia(gP7_P8))

  tO1_O2 = psd(TassO1_O2)
  p7t.append(potencia(tO1_O2))
  aO1_O2 = psd(AassO1_O2)
  p7a.append(potencia(aO1_O2))
  bO1_O2 = psd(BassO1_O2)
  p7b.append(potencia(bO1_O2))
  gO1_O2 = psd(GassO1_O2)
  p7g.append(potencia(gO1_O2))


  tFC5_P7 = psd(TassFC5_P7)
  p8t.append(potencia(tFC5_P7))
  aFC5_P7 = psd(AassFC5_P7)
  p8a.append(potencia(aFC5_P7))
  bFC5_P7 = psd(BassFC5_P7)
  p8b.append(potencia(bFC5_P7))
  gFC5_P7 = psd(GassFC5_P7)
  p8g.append(potencia(gFC5_P7))

  tFC6_P8 = psd(TassFC6_P8)
  p9t.append(potencia(tFC6_P8))
  aFC6_P8 = psd(AassFC6_P8)
  p9a.append(potencia(aFC6_P8))
  bFC6_P8 = psd(BassFC6_P8)
  p9b.append(potencia(bFC6_P8))
  gFC6_P8 = psd(GassFC6_P8)
  p9g.append(potencia(gFC6_P8))

  tP7_O1 = psd(TassP7_O1)
  p10t.append(potencia(tP7_O1))
  aP7_O1 = psd(AassP7_O1)
  p10a.append(potencia(aP7_O1))
  bP7_O1 = psd(BassP7_O1)
  p10b.append(potencia(bP7_O1))
  gP7_O1 = psd(GassP7_O1)
  p10g.append(potencia(gP7_O1))

  tP8_O2 = psd(TassP8_O2)
  p11t.append(potencia(tP8_O2))
  aP8_O2 = psd(AassP8_O2)
  p11a.append(potencia(aP8_O2))
  bP8_O2 = psd(BassP8_O2)
  p11b.append(potencia(bP8_O2))
  gP8_O2 = psd(GassP8_O2)
  p11g.append(potencia(gP8_O2))

  tT7_O1 = psd(TassT7_O1)
  p12t.append(potencia(tT7_O1))
  aT7_O1 = psd(AassT7_O1)
  p12a.append(potencia(aT7_O1))
  bT7_O1 = psd(BassT7_O1)
  p12b.append(potencia(bT7_O1))
  gT7_O1 = psd(GassT7_O1)
  p12g.append(potencia(gT7_O1))

  tT8_O2 = psd(TassT8_O2)
  p13t.append(potencia(tT8_O2))
  aT8_O2 = psd(AassT8_O2)
  p13a.append(potencia(aT8_O2))
  bT8_O2 = psd(BassT8_O2)
  p13b.append(potencia(bT8_O2))
  gT8_O2 = psd(GassT8_O2)
  p13g.append(potencia(gT8_O2))

"""

# 2a. Si se desea para cada electrodo, calcular la PSD
# 2b. Sino, restar la asimetría (hemiIzq-hemiDer) y calcular la PSD

"""


'\n\n# 2a. Si se desea para cada electrodo, calcular la PSD\n# 2b. Sino, restar la asimetría (hemiIzq-hemiDer) y calcular la PSD\n\n'

##Extracción de características EOG

In [ ]:
###Señal ocular vertical

EOGv_parpadeo=[]
EOGv_media=[]
EOGv_var=[]
EOGv_ener=[]

channel = 34 # canal de la señal fisiológica



for i in range (40):
  ojov = X[i-1,channel-1,:]
  ts = len(ojov)/128.
  time = np.linspace(0,ts,len(ojov))
  EOGv_parpadeo.append(tasa(ojov))
  EOGv_media.append(media(ojov))
  EOGv_var.append(varianza(ojov))
  EOGv_ener.append(energia(ojov))


In [ ]:
###Señal ocular horizontal

EOGh_media=[]
EOGh_var=[]
EOGh_ener=[]

channel = 35 # canal de la señal fisiológica

for i in range (40):
  ojoh = X[i-1,channel-1,:]
  EOGh_media.append(media(ojoh))
  EOGh_var.append(varianza(ojoh))
  EOGh_ener.append(energia(ojoh))

#Extracción de características EMG

In [ ]:
### Canal 35 --> Zigomatico

channel = 35 # canal de la señal fisiológica
EMGz_ener=[]
EMGz_media=[]
EMGz_var=[]

for i in range (40):
  muscle = X[i-1,channel-1,:]
  ts = len(muscle)/128.  
  time = np.linspace(0,ts,len(muscle))
  #Caracteristica 1: Energia
  EMGz_ener.append(energia(muscle))
  #Caracterisitica 2: Media y varianza
  EMGz_media.append(media(muscle))
  EMGz_var.append(varianza(muscle))




In [ ]:
### Canal 36 --> Trapecio
channel = 36 # canal de la señal fisiológica
EMGt_ener=[]
EMGt_media=[]
EMGt_var=[]

for i in range (40):
  muscle1 = X[i-1,channel-1,:]
  #Caracteristica 1: Energia
  EMGt_ener.append(energia(muscle1))
  #Caracterisitica 2: Media y varianza
  EMGt_media.append(media(muscle1))
  EMGt_var.append(varianza(muscle1))


#Extracción de características GSR (Respuesta galvánica de la piel)

In [ ]:
###Señal
channel = 37 # canal de la señal fisiológica

GSR_media=[]
GSR_mederi=[]
GSR_meder_neg=[]
GSR_picos= []

for i in range (40):
  señal = X[i-1,channel-1,:]
  ts = len(señal)/128.
  time = np.linspace(0,ts,len(señal))
  # Media
  GSR_media.append(media(señal))
  #Promedio de la derivada
  GSR_mederi.append(media_deriv(señal))
  #Promedio de la derivada negativa
  GSR_meder_neg.append(media_deri_neg(señal))
  ##10 picos densidad espectral
  #Filtro GSR
  """y = lowpass_filter(señal, 2.4, 128, 4)
  #potencia
  pot= psd(y)
  #picos de potencia
  GSR_picos.append(tasa(pot))
  """

print(len(GSR_media))
print(len(GSR_mederi))
print(len(GSR_meder_neg))

40
40
40


In [ ]:
###Numero de minimos locales
"""
def minimos(data):
  peaks, _ = find_peaks(data, prominence=0.01, height=(None,10500))
  plt.plot(data)
  plt.plot(peaks, data[peaks], "o")
  plt.show()
  return (peaks.shape)

min = minimos(señal)
print('Mínimos =',min[0])
"""

'\ndef minimos(data):\n  peaks, _ = find_peaks(data, prominence=0.01, height=(None,10500))\n  plt.plot(data)\n  plt.plot(peaks, data[peaks], "o")\n  plt.show()\n  return (peaks.shape)\n\nmin = minimos(señal)\nprint(\'Mínimos =\',min[0])\n'

#Extracción de características Patrón de respiración (Ritmo cardiaco)

In [ ]:
##Patron respiracion 
channel = 38 # canal de la señal fisiológica

RC_med = []
RC_des_est = []
RC_meder = []


for i in range(40):
  respiration = X[i-1,channel-1,:]
  ts = len(respiration)/128.  
  time = np.linspace(0,ts,len(respiration))
  #Media
  RC_med.append(media(respiration))
  #Deviación Estandar
  RC_des_est.append(desvi_estand(respiration))
  #Promedio derivada
  RC_meder.append(media_deriv(respiration))
  


#Extracción de características Pletismógrafo (presión sanguínea)

In [1]:

channel = 39 # canal de la señal fisiológica

PS_med = []
PS_destand = []
PS_psd = []
PS_psdfp1 = []
PS_psdfp2 = []
PS_psdfp3 = []

for i in range(40):
  presion = X[i-1,channel-1,:]
  ts = len(presion)/128.  
  time = np.linspace(0,ts,len(presion))
  #Caract 1: Media
  PS_med.append(media(presion))
  #Caract 2: Desviacion estandar
  PS_destand.append(desvi_estand(presion))
  #Caract 3: Densidad espectral
  PS_psd.append(psd(presion))
  ## Caract 4: DEP en las diferentes bandas
  #Filtrado Banda (0.1Hz- 0.2Hz) 
  p1 = passband_filter(presion, 0.1, 0.2, 128, 4)
  psd1=psd(p1)
  PS_psdfp1.append(potencia(psd1))
  ## Filtrado Banda (0.2Hz- 0.3Hz) 
  p2 = passband_filter(presion, 0.2, 0.3, 128, 4)
  psd2=psd(p2)
  PS_psdfp2.append(potencia(psd2))
  ## Filtrado Banda (0.3Hz- 0.4Hz) 
  p3 = passband_filter(presion, 0.3, 0.4, 128, 4)
  psd3=psd(p3)
  PS_psdfp3.append(potencia(psd3))




NameError: ignored

#Extracción de características Temperatura de la piel

In [ ]:
channel = 40 # canal de la señal fisiológica


TemP_med= []
TemP_mederiv= []
TemP_psdf1 = []
TemP_psdf2 = []

for i in range(40):
  temperature = X[i-1,channel-1,:]
  ts = len(temperature)/128.  
  time = np.linspace(0,ts,len(temperature))
  #Caract 1: Media
  TemP_med.append(media(temperature))
  #Caract 2: Promedio derivada
  TemP_mederiv.append(media_deriv(temperature))
  # Caract 3: Densidad de potencia Espectral para cada sub-banda
  #Banda 0Hz-0.1Hz
  f1 = filtro_soslp(temperature, 6, 0.1)
  psd1=psd(f1)
  TemP_psdf1.append(potencia(psd1))
  #Banda 0.1Hz-0.2Hz
  f2 = filtro_sospb(temperature, 6, 0.1, 0.2)
  psd2=psd(f2)
  TemP_psdf2.append(potencia(psd2))



## Etiquetas

In [ ]:
labels = db['labels']

Valence = labels[:,0]
Arousal = labels[:,1]


negA = np.logical_and(Arousal>=1,Arousal<=3)
negV = np.logical_and(Valence>=1,Valence<=3)
neutralA = np.logical_and(Arousal>3,Arousal<7)
neutralV = np.logical_and(Valence>3,Valence<7)
posA = Arousal>=7
posV = Valence>=7
Etiquetas = np.zeros((40,1),dtype=int)
# Emocion Negativa negA y negV
Etiquetas[np.where(np.logical_and(negA,negV))] = 1
# Emocion Neutral neutralA y neutralV
Etiquetas[np.where(np.logical_and(neutralA,neutralA))] = 2
# Emocion Positiva posA y posV
Etiquetas[np.where(np.logical_and(posA,posV))] = 3
# Emocion Stress posA y negV
Etiquetas[np.where(np.logical_and(posA,negV))] = 4
# Emocion Calm negA y posV
Etiquetas[np.where(np.logical_and(negA,posV))] = 5
# Emocion Tristeza neutralA y negV
Etiquetas[np.where(np.logical_and(neutralA,negV))] = 6
# Emocion Satisfecho (Pleasant) neutralA y posV
Etiquetas[np.where(np.logical_and(neutralA,posV))] = 7
# Emocion Tranquilidad negA y neutralV
Etiquetas[np.where(np.logical_and(negA,neutralV))] = 8
# Emocion Sorpresa posA y neutralV
Etiquetas[np.where(np.logical_and(posA,neutralV))] = 9



Eti = Etiquetas.reshape(40)

print(Eti)


Etiquetas_name = []

for i in range (0, len(Eti)):

  if (Eti[i] == 1):
    Etiquetas_name.append('Negativa')

  elif (Eti[i] == 2):
    Etiquetas_name.append("Neutral")

  elif (Eti[i] == 3):
    Etiquetas_name.append("Positivo")

  elif (Eti[i] == 4):
    Etiquetas_name.append("Estres")

  elif (Eti[i] == 5):
    Etiquetas_name.append("Calma")

  elif (Eti[i] == 6):
    Etiquetas_name.append("Tristeza")

  elif (Eti[i] == 7):
    Etiquetas_name.append("Satisfech")

  elif (Eti[i] == 8):
    Etiquetas_name.append("Tranquilidad")

  elif (Eti[i] == 9):
    Etiquetas_name.append("Sorpresa")

print(Etiquetas_name)

[2 3 2 2 3 2 7 2 2 9 2 2 2 2 2 2 8 2 7 7 2 2 8 2 2 8 2 2 2 2 2 2 2 2 2 2 9
 2 3 2]
['Neutral', 'Positivo', 'Neutral', 'Neutral', 'Positivo', 'Neutral', 'Satisfech', 'Neutral', 'Neutral', 'Sorpresa', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Tranquilidad', 'Neutral', 'Satisfech', 'Satisfech', 'Neutral', 'Neutral', 'Tranquilidad', 'Neutral', 'Neutral', 'Tranquilidad', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Neutral', 'Sorpresa', 'Neutral', 'Positivo', 'Neutral']


## Concatenando las señales... 

In [ ]:
df = pd.DataFrame({"EOGv_parpadeo": EOGv_parpadeo, "EOGv_media": EOGv_media, "EOGv_varianza": EOGv_var,"EOGv_energia": EOGv_ener,
                   "EOGh_energia": EOGh_ener, "EOGh_media": EOGh_media, "EOGh_varianza": EOGh_var,
                  "EMGz_energia": EMGz_ener,"EMGz_media": EMGz_media, "EMGz_varianza": EMGz_var, 
                  "EMGt_energia": EMGt_ener, "EMGt_media": EMGt_media, "EMGt_varianza": EMGt_var,
                  "GSR_media": GSR_media, "GSR_media_deriv": GSR_mederi, "GSR_med_neg": GSR_meder_neg,
                  "RC_media": RC_med, "RC_media_deriv": RC_meder, "RC_desvi_est": RC_des_est,
                  "PS_media": PS_med, "PS_desvi_est": PS_destand, "PS_potencia1": PS_psdfp1, "PS_potencia2": PS_psdfp2, "PS_potencia3": PS_psdfp3,
                  "Temp_media": TemP_med, "Temp_media_deri": TemP_mederiv, "Temp_potencia1": TemP_psdf1, "Temp_potencia2": TemP_psdf2,
                  "EEG_pot_teta1": p1t, "EEG_pot_alfa1": p1a, "EEG_pot_beta1": p1b, "EEG_pot_gamma1": p1g,
                  "EEG_pot_teta2": p2t, "EEG_pot_alfa2": p2a, "EEG_pot_beta2": p2b, "EEG_pot_gamma2": p2g,
                  "EEG_pot_teta3": p3t, "EEG_pot_alfa3": p3a, "EEG_pot_beta3": p3b, "EEG_pot_gamma3": p3g,
                  "EEG_pot_teta4": p4t, "EEG_pot_alfa4": p4a, "EEG_pot_beta4": p4b, "EEG_pot_gamma4": p4g,
                  "EEG_pot_teta5": p5t, "EEG_pot_alfa5": p5a, "EEG_pot_beta5": p5b, "EEG_pot_gamma5": p5g,
                  "EEG_pot_teta6": p6t, "EEG_pot_alfa6": p6a, "EEG_pot_beta6": p6b, "EEG_pot_gamma6": p6g,
                  "EEG_pot_teta7": p7t, "EEG_pot_alfa7": p7a, "EEG_pot_beta7": p7b, "EEG_pot_gamma7": p7g,
                  "EEG_pot_teta8": p8t, "EEG_pot_alfa8": p8a, "EEG_pot_beta8": p8b, "EEG_pot_gamma8": p8g,
                  "EEG_pot_teta9": p9t, "EEG_pot_alfa9": p9a, "EEG_pot_beta9": p9b, "EEG_pot_gamma9": p9g,
                  "EEG_pot_teta10": p10t, "EEG_pot_alfa10": p10a, "EEG_pot_beta10": p10b, "EEG_pot_gamma10": p10g,
                  "EEG_pot_teta11": p11t, "EEG_pot_alfa11": p11a, "EEG_pot_beta11": p11b, "EEG_pot_gamma11": p11g,
                  "EEG_pot_teta12": p12t, "EEG_pot_alfa12": p12a, "EEG_pot_beta12": p12b, "EEG_pot_gamma12": p12g,
                  "EEG_pot_teta13": p13t, "EEG_pot_alfa13": p13a, "EEG_pot_beta13": p13b, "EEG_pot_gamma13": p13g,
                   "Etiquetas" : Etiquetas_name
                   })

#df.index.name= #= ["Video 1", "Video 2",""] 
df

EOGv_parpadeo  EOGv_media  ...  EEG_pot_gamma13     Etiquetas
0              29  221.166114  ...         9.069169       Neutral
1               1  675.374102  ...        11.539844      Positivo
2            1597 -785.260431  ...        14.353668       Neutral
3              39  109.333299  ...        17.525827       Neutral
4            1843  -41.617158  ...         6.500680      Positivo
5            1561  -21.045649  ...         5.323746       Neutral
6              32  314.964769  ...        20.282041     Satisfech
7              74   32.836724  ...        15.722269       Neutral
8              34  303.097991  ...         8.735848       Neutral
9             288   14.231625  ...        12.998800      Sorpresa
10           1686  -38.326776  ...         7.578970       Neutral
11             59   69.912650  ...        26.240206       Neutral
12           1550 -262.884960  ...        11.789187       Neutral
13           1474 -128.121321  ...        12.252310       Neutral
14           1507 -410.994889  ...         9.487819       Neutral
15             12  545.498422  ...        17.633819       Neutral
16           1601 -139.053813  ...        26.980798  Tranquilidad
17              1  826.738142  ...        11.257858       Neutral
18             46   92.553234  ...         7.926700     Satisfech
19           1698 -360.690998  ...        21.183525     Satisfech
20           1732  -68.263354  ...         6.676422       Neutral
21             35  419.228603  ...         5.371484       Neutral
22            332   16.752808  ...        19.369652  Tranquilidad
23             51  110.537001  ...        13.779019       Neutral
24             64   31.633673  ...         5.798748       Neutral
25           1725 -491.025048  ...         9.717684  Tranquilidad
26           1524 -599.636429  ...        12.331091       Neutral
27            181   14.219341  ...         5.453681       Neutral
28             33  230.009369  ...         6.608497       Neutral
29             45   87.100071  ...         4.473309       Neutral
30             36   42.765785  ...         8.978637       Neutral
31           1661 -157.243376  ...        22.583892       Neutral
32           1524    0.771655  ...         9.785752       Neutral
33             46   27.289304  ...         6.586640       Neutral
34           1666 -101.057675  ...        19.776831       Neutral
35           1820  -36.463134  ...         7.619613       Neutral
36             97   23.958610  ...         8.590786      Sorpresa
37             39   87.418597  ...         6.760034       Neutral
38             93   31.562925  ...        16.521470      Positivo
39             15  481.820724  ...        12.337685       Neutral

[40 rows x 81 columns]

In [ ]:
df.shape

(40, 75)